ewatercycle/forcing.py

```diff
@@ -0,0 +1,341 @@
+from esmvalcore.experimental import get_recipe
+from pathlib import Path
+
+FORCINGS = {
+    'ERA5': {
+        'dataset': 'ERA5',
+        'project': 'OBS6',
+        'tier': 3,
+        'type': 'reanaly',
+        'version': 1
+    },
+    'ERA-Interim': {
+        'dataset': 'ERA-Interim',
+        'project': 'OBS6',
+        'tier': 3,
+        'type': 'reanaly',
+        'version': 1
+    },
+}
+
+
+def update_hype(
+    data: dict,
+    *,
+    forcings: list = None,
+    startyear: int = None,
+    endyear: int = None,
+    shapefile: str = None,
+):
+    """
+    Update hype recipe data in-place.
+    
+    Parameters
+    ----------
+    forcings : list
+        List of forcings to use
+    startyear : int
+        Start year for the observation data
+    endyear : int
+        End year for the observation data
+    shapefile : str
+        Name of the shapefile to use.
+    """
+    preproc_names = ('preprocessor', 'temperature', 'water')
+
+    if shapefile is not None:
+        for preproc_name in preproc_names:
+            data['preprocessors'][preproc_name]['extract_shape'][
+                'shapefile'] = shapefile
+
+    if forcings is not None:
+        datasets = [FORCINGS[forcing] for forcing in forcings]
+        data['datasets'] = datasets
+
+    variables = data['diagnostics']['hype']['variables']
+    var_names = 'tas', 'tasmin', 'tasmax', 'pr'
+
+    if startyear is not None:
+        for var_name in var_names:
+            variables[var_name]['start_year'] = startyear
+
+    if endyear is not None:
+        for var_name in var_names:
+            variables[var_name]['end_year'] = endyear
+
+    return data
+
+
+def update_lisflood(
+    data: dict,
+    *,
+    forcings: list = None,
+    startyear: int = None,
+    endyear: int = None,
+    shapefile: str = None,
+    extract_region: dict = None,
+):
+    """
+    Update lisflood recipe data in-place.
+
+    Parameters
+    ----------
+    forcings : list
+        List of forcings to use
+    startyear : int
+        Start year for the observation data
+    endyear : int
+        End year for the observation data
+    basin : str
+        Name of the basin to use. Defines the shapefile.
```

Comment:
It would be nice to make this more uniform accross all models. Either pass the basin name, or the shapefile. Both have advantages or disadvantages, but I'm inclined towards the latter as it gives more flexibility for the user to add his own experiments.

BTW I think we agreed (once upon a time) that we prefer the term 'catchment' over 'basin'. See https://github.com/eWaterCycle/era5-comparison/blob/master/CONTRIBUTING.md#naming. This is not a criticism, but just a heads up if you ever encounter the term basin you might want to realize this is an ancient artefact of an old age....

Reply:
I'm fine with either, but I understood from Niels that this is in debate.